packages/api/src/controllers/v1/agit/mod.rs

```diff
@@ -29,47 +31,65 @@ impl AgitControllerV1 {
         Self { repo, pool }
     }
 
-    pub fn route(pool: sqlx::PgPool) -> Result<by_axum::axum::Router> {
+    pub fn route(pool: sqlx::PgPool) -> Result<Router> {
         let ctrl = Self::new(pool.clone());
-        Ok(by_axum::axum::Router::new()
-            .route("/:id", post(Self::act_by_id).get(Self::get))
-            .route("/", post(Self::act).get(Self::list))
+        Ok(Router::new()
+            .route("/:id", get(Self::get).post(Self::act_by_id))
+            .route("/", get(Self::list).post(Self::act))
             .with_state(ctrl)
             .nest(
                 "/:agit_id/collections",
                 CollectionControllerV1::route(pool.clone())?,
             ))
     }
-}
 
-impl AgitControllerV1 {
     pub async fn list(
         State(ctrl): State<AgitControllerV1>,
         Extension(claim): Extension<Option<Authorization>>,
         Query(q): Query<AgitParam>,
     ) -> Result<Json<Vec<AgitSummary>>> {
-        //TODO: Add Listing Agits
         tracing::debug!("list agits");
-        Ok(Json(vec![]))
+        let query = AgitQuery {
+            size: 10,
+            bookmark: None,
+        };
+
+        match ctrl.repo.find(&query).await {
+            Ok(agits) => Ok(Json(agits.items)), 
+            Err(e) => Err(ServiceError::Unknown(format!("Failed to retrieve agits: {}", e))),
+        }
     }
+
     pub async fn get(
         State(ctrl): State<AgitControllerV1>,
         Extension(claim): Extension<Option<Authorization>>,
         Path(AgitPathParam { id }): Path<AgitPathParam>,
     ) -> Result<Json<Agit>> {
-        tracing::debug!("get agit {id}");
-        Ok(Json(Agit::default()))
+        tracing::debug!("get agit by id {}", id);
+        let id_action = AgitReadAction { 
```

Comment:
Here, we don't need to handle ReadAction.
ReadAction is for `GET /` uri.

Refer to template how to compose basic outline (https://github.com/biyard/voice-korea/blob/main/packages/api/src/controllers/v2/organizations/_id/members.rs#L219) @dotandev